kivy/uix/image.py

```diff
@@ -138,18 +145,22 @@ def get_image_ratio(self):
         :class:`~kivy.properties.ColorProperty`.
     '''
 
-    allow_stretch = BooleanProperty(False)
+    allow_stretch = BooleanProperty(False, deprecated=True)
     '''If True, the normalized image size will be maximized to fit in the image
     box. Otherwise, if the box is too tall, the image will not be
     stretched more than 1:1 pixels.
 
     .. versionadded:: 1.0.7
 
+    .. deprecated:: 2.2.0
+        :attr:`allow_stretch` have been deprecated. Please use `fit_mode`
+        instead.
+
     :attr:`allow_stretch` is a :class:`~kivy.properties.BooleanProperty` and
     defaults to False.
     '''
 
-    keep_ratio = BooleanProperty(True)
+    keep_ratio = BooleanProperty(True, deprecated=True)
```

Comment:
`keep_ratio` has been deprecated, but is in use from Kivy widgets.
That will show a deprecation warning in the console. Better to migrate these widgets as well (even in a different PR, just after this one is merged, but before 2.2.0 release).

E.g.

https://github.com/kivy/kivy/blob/e2c8f0753337cd693ae5b34de0b3bd3a481d725c/kivy/data/style.kv#L76

(We also have a few examples which are using `keep_ratio`, but we can migrate them later, even after 2.2.0)

Reply:
Hi @misl6! That makes sense! I think this migration can be done in another PR.